src/Components/Filter/Filter.js

```diff
@@ -0,0 +1,208 @@
+import React from "react";
+import { categories } from "../../backend/db/categories";
+import { products } from "../../backend/db/products";
+import { useProduct } from "../../contexts/ProductProvider";
+
+export const getSortedData = (state, productList) => {
+  if (state.sortBy === "PRICE_HIGH_TO_LOW") {
+    return [...productList].sort((a, b) => b.price - a.price);
+  }
+  if (state.sortBy === "PRICE_LOW_TO_HIGH") {
+    return [...productList].sort((a, b) => a.price - b.price);
+  }
+  return productList;
+};
+
+export const filterByRate = (state, productList) => {
+  let data = [...productList];
+  if (state.filterBy === "1STAR_AND_ABOVE") {
+    data = [...productList].filter((product) => product.rate >= 1);
```

Comment:
Good that you destructure. But as you know `filter` itself creates a new array. it is not inplace like sort. So, even if you remove it it will still be immutable

Reply:
Done